device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/DeviceClientConfig.java

```diff
@@ -31,7 +31,10 @@
     private static final int DEFAULT_HTTPS_READ_TIMEOUT_MILLIS = 240000;
     private static final int DEFAULT_HTTPS_CONNECT_TIMEOUT_MILLIS = 0; //no connect timeout
 
-    private static final int DEFAULT_AMQP_OPEN_AUTHENTICATION_SESSION_TIMEOUT_IN_SECONDS = 20;
+    // authentication session timeout is public because a multiplexed connection needs this default if no devices
+    // were registered prior to opening the connection. No device sessions would be opened in that case though, so
+    // the default device session timeout can stay private.
+    public static final int DEFAULT_AMQP_OPEN_AUTHENTICATION_SESSION_TIMEOUT_IN_SECONDS = 20;
     private static final int DEFAULT_AMQP_OPEN_DEVICE_SESSIONS_TIMEOUT_IN_SECONDS = 60;
```

Comment:
q - for a non-multiplexed device, what is the difference between an auth session timeout and a device session timeout?

Reply:
The former is how long the SDK should wait for the CBS session and its links to open, and the latter is how long the SDK should wait for the worker links (twin/telemetry/methods) to open. In the multiplexing case, the authentication session timeout is considered a "Multiplexing level" option, whereas the device session timeout can be set per registered device client.